content/docs/iac/build-with-pulumi/local-packages.md

```diff
@@ -0,0 +1,164 @@
+
+The `pulumi package add` command can accept the package source in multiple ways:
+
+#### From a Plugin Reference
+
+```bash
+pulumi package add PLUGIN[@VERSION]
+```
+
+This attempts to resolve a resource plugin, installing it on-demand similarly to:
+
+```bash
+pulumi plugin install resource PLUGIN [VERSION]
+```
+
+#### From a Local Path
+
+```bash
+pulumi package add ./my-provider
+```
+
+This executes the provider binary to extract its package schema.
+
+#### From a Schema File
+
+```bash
+pulumi package add ./my/schema.json
+```
+
+When the path has a `.json`, `.yml`, or `.yaml` extension, the Pulumi package schema is read directly from it. For details on the structure and syntax of Pulumi package schemas, refer to the [Schema Reference](/docs/iac/using-pulumi/pulumi-packages/schema/). Understanding the schema format is particularly important if you're creating your own custom packages or components.
+
+#### From a Git Repository
+
+```bash
+pulumi package add example.org/org/repo.git/path[@version]
+```
+
+This clones the repo and executes the source. The version can be a tag (in semver format) or a Git commit hash.
+
+### Example: Using a Terraform Provider
+
+One of the most common uses of local packages is with the `terraform-provider` to access any Terraform provider:
+
+```bash
+pulumi package add terraform-provider hashicorp/random
+```
+
+This will generate a local SDK for Hashicorp's random provider, which you can then use in your project:
+
+```typescript
+import * as random from "random";
+
+new random.Pet("my-pet");
+```
+
+## Best Practices for Local Packages
+
+When working with local packages, consider these best practices:
+
+### Managing Generated SDKs
+
+#### Option 1: Don't check in the generated SDK (recommended)
+
+- Add the generated SDK directory to your `.gitignore` file
+- Document the process for other developers to regenerate the SDK
+- Include the `pulumi package add` command in your setup instructions
+
+#### Option 2: Check in the generated SDK
+
+- Pros: Ensures consistent builds and eliminates the need for regeneration
+- Cons: Increases repository size and can lead to merge conflicts
+
+### Updating Local Packages
+
+To update a local package:
+
+1. Re-run the `pulumi package add` command with the updated source
+2. This will regenerate the SDK with the latest schema
+3. Update your imports and code as needed if there are breaking changes
+
+For packages from Git repositories, specify a version tag or commit hash to control which version is used.
+
+### Versioning Considerations
+
+- Consider using Git tags or specific commit hashes when referencing repositories to ensure reproducible builds
+- Document the version of the source provider you're using
+- Be aware that changes to the underlying provider can introduce breaking changes
+
+## Using Generated SDKs
+
+After adding a local package, follow the printed instructions to link it into your project:
+
+1. For Node.js/TypeScript projects:
```

Comment:
Since these aren't really ordered, I wouldn't use a numbered list here.

It's also a prime opportunity to use code choosers.